internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -540,6 +540,20 @@ func (c *container) mountGeneric(mnt *mount.Point) (err error) {
 		}
 	}
 	_, err = c.rpcOps.Mount(source, dest, mnt.Type, flags, optsString)
+	// when using user namespace we always try to apply mount flags with
+	// remount, then if we get a permission denied error, we continue
+	// execution by ignoring the error and warn user if the bind mount
+	// need to be mounted read-only
+	if remount && err != nil && c.userNS {
+		if err.Error() == syscall.Errno(syscall.EPERM).Error() {
```

Comment:
```suggestion
		// this needs to compare error strings because we are dropping
		// the original error value in the RPC call. In order to keep
		// the original error values, we would have to add them to the
		// corresponding struct passed over RPC and add the necessary
		// GOB encoding / decoding functions.
		if err.Error() == syscall.Errno(syscall.EPERM).Error() {
```